feedback_prompt_learning/search_algo/mcts.py

```
@@ -0,0 +1,658 @@
+        self.logger.info("="*80)
+        self.logger.info(f"Total iterations: {len(self.trace_in_each_iter)}")
+        self.logger.info(f"Total nodes created: {len(self.all_nodes)}")
+        self.logger.info(f"\nBest Q path (iteration {qs_rank[0]}):")
+        self.logger.info(f"  Depth: {len(best_q_path)-1}")
+        self.logger.info(f"  Mean Q: {np.mean([n.Q for n in best_q_path]):.4f}")
+        self.logger.info(f"  Mean Reward: {np.mean([n.reward for n in best_q_path]):.4f}")
+        self.logger.info(f"\nBest Reward path (iteration {rewards_rank[0]}):")
+        self.logger.info(f"  Depth: {len(best_reward_path)-1}")
+        self.logger.info(f"  Mean Q: {np.mean([n.Q for n in best_reward_path]):.4f}")
+        self.logger.info(f"  Mean Reward: {np.mean([n.reward for n in best_reward_path]):.4f}")
+        self.logger.info(f"\nSelected node (highest reward in best_reward_path):")
```

Ruff (0.14.8)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)